rpc/filters.go

```diff
@@ -0,0 +1,84 @@
+package rpc
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+)
+
+/*
+	- Filter functions derived from go-ethereum
+	Used to set the criteria passed in from RPC params
+*/
+
+// Filter can be used to retrieve and filter logs.
+type Filter struct {
```

Comment:
Reason for excluding range filter parameters and bloom filter matcher? just leaving out just for this PR to be added later or is there another reason?

Reply:
Exactly, it will be added to in subsequent PR that makes use of the bloom filter matcher...the linter was just complaining during the `structcheck` that these fields were unused.

Reply:
Gotcha! Just was curious since the range filter functionality was somewhat included in this PR (Although maybe not the focus) so as long as we remember to come back to this